obi_one_v2/variables.tf

```diff
@@ -78,19 +63,35 @@ variable "container_port" {
   type        = number
 }
 
-variable "mounted_volume_name" {
-  description = "Name of the volume to be mounted"
-  type        = string
-}
-
-variable "mounted_volume_host_path" {
-  description = "Path of the mounted volume on the host"
-  type        = string
-}
-
-variable "mounted_volume_container_path" {
-  description = "Path of the mounted volume in the container"
+variable "mount_base_dir" {
+  description = "Base directory for all the mountpoints in ECS, used to prefix for both volume_host_path and volume_container_path."
   type        = string
+  validation {
+    condition     = can(regex("^(/.*)?$", var.mount_base_dir))
+    error_message = "mount_base_dir must be empty or start with /"
+  }
+}
+
+variable "mount_buckets" {
+  description = "List of buckets to be mounted in EC2 and accessible in ECS."
+  type = list(object({
+    bucket_name           = string # Name of the bucket containing data to be mounted
+    bucket_region         = string # Region of the bucket containing data to be mounted
+    bucket_prefix         = string # Prefix for data to be mounted, must end with / if not empty
+    volume_name           = string # Name of the volume to be mounted
+    volume_host_path      = string # Path of the mounted volume on the host, should be set to /{storage_type}/{bucket_prefix}
+    volume_container_path = string # Path of the mounted volume in the container, should be set to /{storage_type}/{bucket_prefix}
+  }))
+  validation {
```

Comment:
neat, never touched the validation stuff

is it worth checking for duplicates?

Reply:
Yeah it's safer, I improved the validation by checking for uniqueness in volume_name, volume_host_path, and volume_container_path